civic_scraper/platforms/legistar/scraper.py

```diff
@@ -1,31 +1,59 @@
+import re
 import civic_scraper
 from civic_scraper import base
 from civic_scraper.base.asset import Asset, AssetCollection
 from civic_scraper.base.cache import Cache
 
 from legistar.events import LegistarEventsScraper
 
-from datetime import datetime
+from datetime import datetime, time
 from pathlib import Path
 from urllib.parse import urlparse, parse_qs
+from warnings import showwarning
 
 # Scrape today's agendas and minutes from a Legistar site
 class LegistarSite(base.Site):
 
-    def __init__(self, base_url, cache=Cache(), parser_kls=None, timezone=None):
+    def __init__(self, base_url, event_info_key='Meeting Details', cache=Cache(), parser_kls=None, timezone=None):
         super().__init__(base_url, cache, parser_kls)
         self.legistar_instance = urlparse(base_url).netloc.split('.')[0]
         self.timezone = timezone
+        self.event_info_key = event_info_key
 
     def create_asset(self, event, scraper):
-        # get date and time of event
-        meeting_datetime = " ".join((event['Meeting Date'], event['Meeting Time']))
+        location_info = None
+        if self.event_info_key == 'Meeting Details':
```

Comment:
this should be 

```
detail_info = event[self.event_info_key]
```

and all the other differences should be other arguments that you pass to this classes init method. you could pass it like a dict if you wanted

```python 
def __init__(self, base_url, event_info_key='Meeting Details', event_info_keys = {'meeting_date_info': 'Meeting Date', 'meeting_time_info': 'Meeting Time', 'meeting_location_info': 'Meeting Location'}, cache=Cache(), parser_kls=None, timezone=None):
    ...
    detail_info[event_info_key]
    date_info = date_info[event_info_keys[meeting_date_info]]
    ...
```

basically, you want to avoid hard coding these switch statements and change them into configuration options.